core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -624,6 +625,7 @@ public RelRoot convertQuery(
       final boolean needsValidation,
       final boolean top) {
     final boolean unwrapMeasures = !validator().config().embeddedQuery();
+    query = implementArithmetic(validator().config().conformance(), query);
```

Comment:
There is another pattern for introducing operators based on conformance that doesn't require multiple passes and modifications of the AST. See `SqlStdOperatorTable` and in particular the changes around CALCITE-5747, CALCITE-6730, etc.